llvm/lib/Target/AArch64/AArch64ExpandImm.cpp

```diff
@@ -518,6 +518,14 @@ static inline void expandMOVImmSimple(uint64_t Imm, unsigned BitSize,
     Insn.push_back({ Opc, Imm16,
                      AArch64_AM::getShifterImm(AArch64_AM::LSL, Shift) });
   }
+
+  // Now, we get 16-bit divided Imm. If high and low bits are same in
+  // 32-bit, there is an opportunity to reduce instruction.
+  if (Insn.size() > 2 && (Imm >> 32) == (Imm & UINT_MAX)) {
```

Comment:
I would use 0xffffffff instead of UINT_MAX. They are likely always the same value, but the ff's is more explicit about the size.

Reply:
got it!